### PR TITLE
Anchor link whitelist matching at TLD

### PR DESCRIPTION
**Pull request by robsdedude**
As explained in #2212, the current implementation of the links white list is still vulnerable in a sense that it, probably for most users unexpectedly, permits some links. 

Examples and tests can be found here: https://github.com/PhantomBot/PhantomBot/pull/2212#issuecomment-603141513

To clarify how this works: It makes sure that the TLD (e.g. `.com`) of the URL is part of the white-listed link match.

### CODE REVIEW

**Review by gmt2001 — APPROVED** [review]

File: javascript-source/core/chatModerator.js [hunk 1]
```diff
@@ -494,10 +495,24 @@
      * @param {string} message
      */
     function checkWhiteList(message) {
+        function checkLink(link, whiteListItem) {
+            var baseLink = link.match(/[^.]*[^/]*/)[0];
+            var itemRe = RegExp(whiteListItem.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'), 'g');
```
Comment (BrandenB) [inline javascript-source/core/chatModerator.js]: I'm not sure if in NodeJS not using `new` on a class works, but in Rhino it will cause an error for sure.

**Review by gmt2001 — CHANGES_REQUESTED** [review]
Needs `new` for RegExp

**Comment by robsdedude** [comment]
> Needs `new` for RegExp

oopsie. Done.

**Review by gmt2001 — APPROVED** [review]

**Comment by robsdedude** [comment]
@BrandenB, anything else you want me to do?

**Review by BrandenB — APPROVED** [review]

**Pull request merged**